blocks/blocks_test.go

```diff
@@ -14,3 +20,77 @@ func TestBlocksBasic(t *testing.T) {
 	// Test some data
 	NewBlock([]byte("Hello world!"))
 }
+
+func TestData(t *testing.T) {
+	data := []byte("some data")
+	block := NewBlock(data)
+
+	if !bytes.Equal(block.Data(), data) {
+		t.Error("data is wrong")
+	}
+}
+
+func TestHash(t *testing.T) {
+	data := []byte("some other data")
+	block := NewBlock(data)
+
+	hash, err := mh.Sum(data, mh.SHA2_256, -1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !bytes.Equal(block.Multihash(), hash) {
+		t.Error("wrong multihash")
+	}
+}
+
+func TestKey(t *testing.T) {
+	data := []byte("yet another data")
+	block := NewBlock(data)
+	key := block.Key()
+
+	if !bytes.Equal(block.Multihash(), key.ToMultihash()) {
+		t.Error("key contains wrong data")
+	}
+}
+
+func TestManualHash(t *testing.T) {
+	oldDebugState := u.Debug
+	defer (func() {
+		u.Debug = oldDebugState
+	})()
+
+	data := []byte("I can't figure out more names .. data")
+	hash, err := mh.Sum(data, mh.SHA2_256, -1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	u.Debug = false
+	block, err := NewBlockWithHash(data, hash)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !bytes.Equal(block.Multihash(), hash) {
+		t.Error("wrong multihash")
+	}
+
+	data[5] = byte((uint32(data[5]) + 5) % 256) // Transfrom hash to be different
+	block, err = NewBlockWithHash(data, hash)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !bytes.Equal(block.Multihash(), hash) {
+		t.Error("wrong multihash")
+	}
+
+	u.Debug = true
+
+	block, err = NewBlockWithHash(data, hash)
+	if err == nil {
```

Comment:
I'd prefer to explicitly check errors in cases like this. I've had a few instances of tests passing while the code itself was breaking because a different error was being returned